executor/predictor/ready_checker.go

```diff
@@ -2,13 +2,30 @@ package predictor
 
 import (
 	"fmt"
+	"github.com/seldonio/seldon-core/executor/api"
 	"github.com/seldonio/seldon-core/operator/apis/machinelearning.seldon.io/v1"
 	"net"
+	"net/http"
+	"net/url"
+	"strconv"
 )
 
-func Ready(node *v1.PredictiveUnit) error {
+func Ready(protocol string, node *v1.PredictiveUnit) error {
+	switch protocol {
+	case api.ProtocolSeldon:
+		return ReadyHealth(node, "/api/v1.0/health/status")
+	case api.ProtocolTensorflow:
+		return ReadyTCP(node)
```

Comment:
The tensoflow protocol does seem to have a way to query the model status/availability, but seems it's not fully standardised so could be something that can be explored further down the line